trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -45,23 +47,20 @@ def create_project(self, sb_lead_id, x_api_key, project_template_id=None):
             if not project_exist:
                 project_env = self.env['project.project']
                 # Create new project
-                if project_template_id: # Template ID is provided
-                    if type(project_template_id) == int:
-                        template = project_env.search([('id', '=', project_template_id), ('name', 'like', ' (TEMPLATE)')])
-                        if template: # Template exists
-                            project = template.create_project_from_template(sb_lead_id)
-                            project_new = project_env.search([('id', '=', project['res_id'])], limit=1)
-                        else: # Template does not exist, treat as no template ID
-                            project_new = project_env.create({"name": 'SBLead-' + str(sb_lead_id), "sb_lead_id": sb_lead_id})
-                    else: # Template ID is not an Integer, treat as no template ID
+                if self.param_check(input_dict) == 'is_temp':
```

Comment:
is it necessary when all the parameters have been valid already since we have checked all the cases above?